packages/userscript/source/ui/UserInterface.ts

```diff
@@ -72,6 +72,48 @@ export class UserInterface {
     optionsListElement.append(this._optionsUi.element);
     optionsListElement.append(this._filterUi.element);
 
+    // Make _engineUI's expando button hide/show the other option groups
+    // Currently accesses the button via id.
+    let optionsToggle = this._engineUi.element.children("#toggle-items-engine");
+    optionsToggle.on("click", () => {
+      let optionsVisiblity = this._engineUi.toggleOptions();
+      this._bonfireUi.element.toggle(optionsVisiblity);
+      this._spaceUi.element.toggle(optionsVisiblity);
+      this._craftUi.element.toggle(optionsVisiblity);
+      this._unlockUi.element.toggle(optionsVisiblity);
+      this._tradingUi.element.toggle(optionsVisiblity);
+      this._religionUi.element.toggle(optionsVisiblity);
+      this._timeUi.element.toggle(optionsVisiblity);
+      this._timeCtrlUi.element.toggle(optionsVisiblity);
+      this._distributeUi.element.toggle(optionsVisiblity);
+      this._optionsUi.element.toggle(optionsVisiblity);
+      this._filterUi.element.toggle(optionsVisiblity);
+
+      optionsToggle.text(optionsVisiblity ? "-" : "+");
+      optionsToggle.prop(
+        "title",
+        optionsVisiblity ? this._host.i18n("ui.itemsHide") : this._host.i18n("ui.itemsShow")
+      );
+    });
+
+    // collapse all options if the Enable Kitten Scientists label is shift-clicked
+    this._engineUi.element.children("label").on("click", e => {
```

Comment:
I try not to use abbreviated variable names in the code. I know it might seem too verbose, but I'd still appreciate `event` over `e` here.